### PR TITLE
Adding memo.py

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: example/provider/main.py [hunk 1]
```diff
@@ -212,11 +222,46 @@ async def do_requests(api: bxsolana.Provider):
     )
 
 
+async def create_transaction_with_memo(api: bxsolana.Provider):
+    pv1 = "3KWC65p6AvMjvpR2r1qLTC4HVSH4jEFr5TMQxagMLo1o3j4yVYzKsfbB3jKtu3yGEHjx2Cc3L5t8wSo91vpjT63t"
```
Comment (aspin) [inline example/provider/main.py]: are these real keys? should probably pull them from github if so and load them from ENV variables

File: example/provider/main.py [hunk 2]
```diff
@@ -212,11 +222,46 @@ async def do_requests(api: bxsolana.Provider):
...
+
+
 async def do_transaction_requests(api: bxsolana.Provider):
     if not RUN_TRADES:
         print("skipping transaction requests: set by environment")
         return
 
+    print(
+        "creating tx with memo"
+    )
```
Comment (aspin) [inline example/provider/main.py]: need to call actually call `create_transaction_with_memo`

File: bxsolana/transaction/memo.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+from solana.keypair import Keypair
+
+
+BxMemoMarkerMsg = "Powered by bloXroute Trader Api"
+TraderAPIMemoProgram = PublicKey("HQ2UUt18uJqKaQFJhgV9zaTdQxUZjNrsKFgoEDquBkcx")
+
+
+# create_trader_api_memo_instruction generates a transaction instruction that places a memo in the transaction log
+# Having a memo instruction with signals Trader-API usage is required
+def create_trader_api_memo_instruction(msg):
```
Comment (aspin) [inline bxsolana/transaction/memo.py]: file needs type annotation for param + return type

File: bxsolana/transaction/memo.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        msg = BxMemoMarkerMsg
+
+    data = bytes(msg, 'utf-8')
+    instruction = solana_transaction.TransactionInstruction(keys=[], program_id=TraderAPIMemoProgram,
+                                                            data=data)
+
+    return instruction
+
+
+def add_memo(instructions: list[solana_transaction.TransactionInstruction], memo_content: str, blockhash, owner, *private_keys: Keypair):
```
Comment (aspin) [inline bxsolana/transaction/memo.py]: return type annotation + other parameter annotations

File: example/provider/main.py (hunk 1 above)
Comment (aspin) [reply example/provider/main.py]: nvm i see that they're generated. still probably should make these ENV variables though, if people want to test the submission?

File: bxsolana/transaction/__init__.py [hunk 5]
```diff
@@ -12,4 +19,7 @@
     "sign_tx",
     "sign_tx_with_private_key",
     "load_open_orders",
-]
+    "create_trader_api_memo_instruction",
+    "build_fully_signed_txn",
```
Comment (aspin) [inline bxsolana/transaction/__init__.py:22]: think this function doesn't need to be exported

File: example/provider/main.py (hunk 1 above)
Comment (tinystarinagalaxy) [reply example/provider/main.py]: these are generated, we can make it ENV variables, but its an overkill at this point.

File: example/provider/main.py (hunk 2 above)
Comment (tinystarinagalaxy) [reply example/provider/main.py]: i have defined another function `create_transaction_with_memo`  deleted this print statement

File: bxsolana/transaction/memo.py (hunk 3 above)
Comment (tinystarinagalaxy) [reply bxsolana/transaction/memo.py]: done

File: bxsolana/transaction/memo.py (hunk 4 above)
Comment (tinystarinagalaxy) [reply bxsolana/transaction/memo.py]: done

File: bxsolana/transaction/__init__.py (hunk 5 above)
Comment (tinystarinagalaxy) [reply bxsolana/transaction/__init__.py:22]: its used in the example

**Review by aspin — APPROVED** [review]

**Pull request merged**